core/state_transition.go

```diff
@@ -280,9 +312,28 @@ func (st *StateTransition) TransitionDb() (ExecutionResult, error) {
 	if contractCreation {
 		ret, _, st.gas, vmErr = evm.Create(sender, st.data, st.gas, st.value)
 	} else {
-		// Increment the nonce for the next transaction
-		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
+		if !msg.IsAA() {
+			// Increment the nonce for the next transaction
+			st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
+		}
 		ret, st.gas, vmErr = evm.Call(sender, st.to(), st.data, st.gas, st.value)
+		if msg.IsAA() && st.evm.PaygasMode != vm.PaygasNoOp {
+			if vmErr != nil {
+				return ExecutionResult{}, vmErr
+			} else {
+				return ExecutionResult{}, ErrNoPaygas
+			}
+		}
+	}
+	if st.msg.IsAA() {
+		st.gasPrice = st.evm.GasPrice
+		if msg.IsAA() && st.evm.PaygasMode != vm.PaygasNoOp {
```

Comment:
seems this part from 330-336 is not in the eth code? why it's added here?

Reply:
Oops, this seems to be a mistake. thanks for catching, I'll remove it!